### ISSUE TITLE
node_colour_by does not colour nodes

### ISSUE THREAD

**Issue by RiboRings** [microbiome/miaViz#131]
Hi!

I think the `node_colour_by` arg of `plotRowTree` shows an unexpected behaviour. For example the following code generates an image without colours:

```
library(miaViz)
library(mia)
data("Tengeler2020", package = "mia")
tse <- Tengeler2020
plotRowTree(tse, node_colour_by = "Class")
```

![Screenshot 2024-06-19 at 16 19 43](https://github.com/microbiome/miaViz/assets/82203661/b311d347-bdc5-43e4-92b1-c7c227d9b56e)

**Comment by antagomir** [comment]
In what way you think it is unexpected?

**Comment by RiboRings** [comment]
Not sure, shouldn't the edges be coloured by Class?

**Comment by antagomir** [comment]
Ah, yes. @ElySeraidarian do you see anything here?

**Comment by ElySeraidarian** [comment]
I will check this.

**Comment by ElySeraidarian** [comment]
It seems that there is indeed an issue as it considers nodes as NA values 
`plot_out <- .resolve_plot_colours(plot_out,
                                          plot_out$data$colour_by,
                                          colour_by,
                                          fill = point_out$fill,
                                          na.translate = FALSE,
                                          na.value = "red")`
Here by adding `na.value = "red"`, it actually changes the plot like that : 
![image](https://github.com/microbiome/miaViz/assets/167744870/9e1001e7-0ba2-4c61-8cdc-704e032d75cf)

I have not been able to solve the issue yet but I'm working on it.

**Comment by antagomir** [comment]
@ElySeraidarian any updates on this one?

**Comment by ElySeraidarian** [comment]
I have not been able to solve the problem yet even though I can see where it happens but can't understand why. 
https://github.com/microbiome/miaViz/blob/5ba5e7c881566e3675e0d8c974fef05f983ec69f/R/plotTree.R#L1421
```
library(miaViz)
library(mia)
data("Tengeler2020", package = "mia")
tse <- Tengeler2020
plotRowTree(tse, node_colour_by = "Class")
```
Here if we add a print statement of the plot after adding nodes in the if statement, here's what we get :
![image](https://github.com/user-attachments/assets/18a625e8-d8a0-48e6-81d9-37b53641a778)

Whereas, ```plotRowTree(tse, tip_colour_by = "Class")``` will get this figure.
![image](https://github.com/user-attachments/assets/d64a5205-9a61-412e-aac8-1ee163625e9c).
I am still unsure how  it gets the labels for the legend when plotting with tip_colour_by, it should be stored in ```plot_out$data$colour_by``` but this is only used after in ```.plot_resolve_colours()```

I will still give a try but have been quite unsuccessful so far

**Comment by antagomir** [comment]
Right, tips correspond to the final ends end nodes also to the intermediate branch points. 

For many current purposes the `tip_colour_by` should be enough. The names for that come from rowData directly? (each tip (and node) associates with one rowData row).

Could you update the examples so that they show how this can be done for the tips?

Are the links between nodes and rowData rows available? (rowLinks)? Missing links might explain the issue. If links are missing, little can be done afterwards, those links should be added already when the rowTree is integrated.
